syntax = "proto3";

package arm_face_id;

message RegisterRequest {
  bytes face_img = 1;
  string name = 2;
  int64 id = 3;
}

message RegisterResult {
  bytes face_img = 1;
  string name = 2;
  int64 id = 3;
  string last_recognized_date = 4;
}

message RecognizeRequest {
  bytes face_img = 1;
}

message RecognizeResult {
  bytes face_img = 1;
  string name = 2;
  int64 id = 3;
  string last_recognized_date = 4;
  string email = 5;
}

service RpcManager {
  rpc RecognizeFace(RecognizeRequest) returns (RecognizeResult) { }

  rpc Register(RegisterRequest) returns (RegisterResult) { }
}